### PR TITLE
Adjust normalizer & serializer to enable retract mode

### LINKED ISSUE #14042: Adjust normalizer & serializer
(1) Define a switch (for instance, see here: feature-libs/product-configurator/rulebased/components/config)
(2) Define a translatable attribute so that you can customize the text that is shown ("No option selected", "No selection", "None", "No entry" etc.)
(3) Adjust serializer accordingly (if the "No option selected" is set the retractTriggered should be set to true)

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts [hunk 1]
```diff
@@ -760,4 +797,311 @@ describe('OccConfiguratorVariantNormalizer', () => {
...
+      );
+      expect(value.valueDisplay).toEqual(
+        'configurator.attribute.noOptionSelectedMsg'
+      );
+    });
+  });
+
+  describe('addRetractValue', () => {
+    afterEach(() => {
+      MockOccModuleConfig.backend.retractTriggered = true;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Strict mode issue

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts [hunk 2]
```diff
@@ -760,4 +797,311 @@ describe('OccConfiguratorVariantNormalizer', () => {
...
+    });
+
+    it('should not add a retract value to the list of values because the retract mode is not activated', () => {
+      MockOccModuleConfig.backend.retractTriggered = false;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Strict mode issue

File: projects/core/src/occ/config/occ-config.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ export interface BackendConfig {
     baseUrl?: string;
   };
   loadingScopes?: LoadingScopes;
+  retractTriggered?: boolean;
```
Comment (ChristophHi) [inline projects/core/src/occ/config/occ-config.ts]: (1) We should not introduce that in the generic config file
(2) The name is confusing. This is about adding an additional value, which should be reflected in the parameter name

File: projects/core/src/occ/adapters/product/default-occ-product-config.ts [hunk 4]
```diff
@@ -34,5 +34,6 @@ export const defaultOccProductConfig: OccConfig = {
         },
       },
     },
+    retractTriggered: false,
```
Comment (ChristophHi) [inline projects/core/src/occ/adapters/product/default-occ-product-config.ts]: We should not introduce that in the generic config file

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: projects/core/src/occ/adapters/product/default-occ-product-config.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply projects/core/src/occ/adapters/product/default-occ-product-config.ts]: Where should the retract mode be introduced?

File: projects/core/src/occ/config/occ-config.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply projects/core/src/occ/config/occ-config.ts]: What is the better naming according to your point of view?

Comment (ChristophHi) [reply projects/core/src/occ/config/occ-config.ts]: What about 'addRetractOption' ?

File: projects/core/src/occ/adapters/product/default-occ-product-config.ts (hunk 4 above)
Comment (ChristophHi) [reply projects/core/src/occ/adapters/product/default-occ-product-config.ts]: As we just discussed, ProductConfiguratorUISettingsConfig is probably a good place

File: projects/core/src/occ/config/occ-config.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply projects/core/src/occ/config/occ-config.ts]: Done!

File: projects/core/src/occ/adapters/product/default-occ-product-config.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply projects/core/src/occ/adapters/product/default-occ-product-config.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts]: Done!

**Review by ChristophHi — CHANGES_REQUESTED** [review]
This does not work with CPS, the retract value code must not be sent to the backend. CPS responds with

ERROR [hybrisHTTP25] [RequestErrorHandlerImpl] While calling the cps service the following server error occured:{
  "status" : 422,
  "typeId" : null,
  "type" : "unprocessable entity",
  "message" : "Update checks failed",
  "messageDetails" : {
    "area" : "CFG_VALIDATION",
    "id" : "002",
    "created" : "2021-11-03T05:45:20.991Z",
    "severity" : "ERROR",
    "transient" : true,
    "plainMessage" : "Update checks failed",
    "attributes" : [ ],
    "formattedMessage" : "Update checks failed"
  },
  "moreInfo" : null,
  "details" : [ {
    "typeId" : "40003.002",
    "type" : "data_invalid",
    "field" : "CHHI_CON",
    "value" : "###RETRACT_VALUE_CODE###",
    "message" : "Value ###RETRACT_VALUE_CODE### exceeds the maximum length of 1 for characteristic CHHI_CON",
    "messageDetails" : {
      "area" : "CFG_VALIDATION",
      "id" : "150",
      "created" : "2021-11-03T05:45:20.989Z",
      "severity" : "ERROR",
      "transient" : true,
      "plainMessage" : "Value {0} exceeds the maximum length of {1} for characteristic {2}",
      "attributes" : [ {
        "attribute" : "###RETRACT_VALUE_CODE###",
        "type" : "string"
      }, {
        "attribute" : "1",
        "type" : "float"
      }, {
        "attribute" : "CHHI_CON",
        "type" : "string"
      } ],
      "formattedMessage" : "Value ###RETRACT_VALUE_CODE### exceeds the maximum length of 1 for characteristic CHHI_CON"
    },
    "moreInfo" : null
  } ]
}

**Review by ChristophHi — CHANGES_REQUESTED** [review]
Still does not work. It's possible to de-select, but afterwards, no value can be set

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**